File: BookStoreBackEnd/BookStoreBackEnd/DTOs/UserDTO.cs
public class UserDTO
{
    public int UserId { get; set; }
    public string Name { get; set; }
    public DateTime DateOfBirth { get; set; }
    public string PhoneNumber { get; set; }
    public string Username { get; set; }
    public string Email { get; set; }
    public int RoleId { get; set; }
    public string? PasswordHash { get; set; }
}

File: BookStoreFrontEnd/src/app/services/user.dto.ts
export class UserDTO {
  userId: number = 0; // Default to 0
  name?: string; // Ensure `name` is included
  dateOfBirth?: string;
  phoneNumber?: string;
  username?: string;
  email?: string;
  roleId: number = 2; // Default role id
  passwordHash?: string;
  passwordConfirm?: string;

  constructor() {
    this.roleId=2;
  }
}
